packages/devtools_app/lib/src/service/vm_service_wrapper.dart

```diff
@@ -285,9 +290,57 @@ class VmServiceWrapper implements VmService {
         },
       });
     }
+
+    await _addProcessedUrlsToTraceObject(
+        isolateId, resolvedUrlMap, traceObject);
+
     return CpuProfileData.parse(traceObject);
   }
 
+  // Helper function for fetching simplified package urls for each of
+  // the [resolvedUrls] then mapping them to the [traceObject].
+  //
+  // [isolateId] The id which is passed to the getIsolate RPC to load this isolate.
+  // [resolvedUrlMap] A map of where each key is a resolved url that has been extracted from the [traceObject]'s
+  // stack frames. The values of the map will be overriden in this helper.
+  // This is passed as a parameter since [resolvedUrls] can be extracted
+  // while [traceObject] is being created.
+  // [traceObject] A map where the CpuProfileData for each frame is stored.
+  Future<void> _addProcessedUrlsToTraceObject(
+      String isolateId,
+      Map<String, String?> resolvedUrlMap,
+      Map<String, dynamic> traceObject) async {
+    final processedUrlMapping = <String, String>{};
```

Comment:
we should store this mapping as a field on the VmServiceWrapper object. That way we can reuse the processed urls for future recorded profiles. As long as we have the same vm service connection, these urls should persist from request to request.